src/drt/src/pa/FlexPA_prep.cpp

```diff
@@ -2624,6 +2624,37 @@ int FlexPA::getEdgeCost(
   return edge_cost;
 }
 
+std::vector<int> FlexPA::extractAccessPatternFromNodes(
+    const std::vector<FlexDPNode>& nodes,
+    const std::vector<std::pair<frMPin*, frInstTerm*>>& pins,
+    std::set<std::pair<int, int>>& used_access_points,
+    const int max_access_point_size)
+{
+  std::vector<int> access_pattern(pins.size(), -1);
+
+  const int source_node_idx = getFlatIdx(-1, 0, max_access_point_size);
+  const int drain_node_idx = getFlatIdx(pins.size(), 0, max_access_point_size);
+  auto drain_node = &(nodes[drain_node_idx]);
+  int curr_node_idx = drain_node->getPrevNodeIdx();
```

Comment:
Does "drain" have a special meeting here? I the opposite of source is usually "sink" in these contexts

Reply:
I guess sink is the correct word in this context.